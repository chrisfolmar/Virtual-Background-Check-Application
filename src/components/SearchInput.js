import React, { Component } from 'react'

/**
 * A search bar component
 */
 class SearchInput extends Component {

	/**
     * constructor
     * @param {object} props
     */
     constructor(props){
     	super(props);

		/**
         * @type {object}
         * @property {string} fieldVal - searched IP
         */
         this.state = 
         {
         	fieldVal: ""
         };

         this.handleChange = this.handleChange.bind(this);
         this.handleSubmit = this.handleSubmit.bind(this);
     }

	/**
     * handle change event at input form
     * @param {SytheticEvent} event
     */
     handleChange(event) {
     	this.setState({fieldVal: event.target.value});
     }

	/**
     * handle submit form event
     * @param {SytheticEvent} event
     */
     handleSubmit(event) {
     	event.preventDefault()
     	this.props.onSubmission(this.state.fieldVal)
     }

	/**
     * render
     * @return {ReactElement} markup
     */
     render(){

     	var blacklisted = {
     		color: 'black'
     	};

     	return(
     		<header className="masthead text-white text-center">
				<div className="overlay"></div>
					<div className="container">
						<div className="row">
							<div className="col-xl-9 mx-auto">
							<h2 className="mb-5">
								Verify if an IP Address is <span style={blacklisted}>blacklisted</span>
							</h2>
						</div>
						<div className="col-md-10 col-lg-8 col-xl-7 mx-auto">
							<form onSubmit={this.handleSubmit}>
								<div className="form-row">
									<div className="col-12 col-md-9 mb-2 mb-md-0">
											<input 
												type="text" 
												className="form-control form-control-lg" 
												placeholder="Enter IP Address to verify..."
												onChange={this.handleChange}
												value={this.state.fieldVal}
												pattern="\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.|$)){4}\b"
												title="1.2.3.4"
											/>
									</div>
									<div className="col-12 col-md-3">
										<button type="submit" className="btn btn-block btn-lg btn-primary">Search</button>
									</div>
								</div>
							</form>
						</div>
					</div>
				</div>
			</header>
 		);
     }
 }

 export default SearchInput;