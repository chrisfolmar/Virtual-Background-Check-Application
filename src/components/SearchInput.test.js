import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json'
import SearchInput from './SearchInput';
import { spy } from 'sinon';

describe('SearchInput', () => {

	it('renders fully without crashing', () => {
		const component = mount(<SearchInput onSubmission={[Function]} />);
		component.unmount();
	});

	it('full snapshot rendering', () => {
		const component = mount(
			<SearchInput onSubmission={[Function]} />
			);
		expect(mountToJson(component)).toMatchSnapshot();
		component.unmount();
	});

	it('renders submit button', () => {
		const onSubmit = spy();
		const component = mount(<SearchInput onSubmission={onSubmit} />);
		const submitButton = <div className="col-12 col-md-3">
								<button type="submit" className="btn btn-block btn-lg btn-primary">Search</button>
							</div>;
		expect(component.contains(submitButton)).toEqual(true);
		component.unmount();
	});

	it('handles submit with submit event on form', () => {
		const onSubmit = spy();
		const component = mount(<SearchInput onSubmission={onSubmit} />);
		const form = component.find('form');
		form.simulate('submit');
		expect(onSubmit.called).toEqual(true);
		component.unmount();
	});

	it('handles submit with click event on button', () => {
		const onSubmit = spy();
		const component = mount(<SearchInput onSubmission={onSubmit} />);
		const button = component.find('button');
		button.simulate('submit');
		expect(onSubmit.called).toEqual(true);
		component.unmount();
	});

});

