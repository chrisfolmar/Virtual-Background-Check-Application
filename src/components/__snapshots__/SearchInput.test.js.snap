// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput full snapshot rendering 1`] = `
<SearchInput
  onSubmission={
    Array [
      [Function],
    ]
  }
>
  <header
    className="masthead text-white text-center"
  >
    <div
      className="overlay"
    />
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-xl-9 mx-auto"
        >
          <h2
            className="mb-5"
          >
            Verify if an IP Address is 
            <span
              style={
                Object {
                  "color": "black",
                }
              }
            >
              blacklisted
            </span>
          </h2>
        </div>
        <div
          className="col-md-10 col-lg-8 col-xl-7 mx-auto"
        >
          <form
            onSubmit={[Function]}
          >
            <div
              className="form-row"
            >
              <div
                className="col-12 col-md-9 mb-2 mb-md-0"
              >
                <input
                  className="form-control form-control-lg"
                  onChange={[Function]}
                  pattern="\\\\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\\\.|$)){4}\\\\b"
                  placeholder="Enter IP Address to verify..."
                  title="1.2.3.4"
                  type="text"
                  value=""
                />
              </div>
              <div
                className="col-12 col-md-3"
              >
                <button
                  className="btn btn-block btn-lg btn-primary"
                  type="submit"
                >
                  Search
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </header>
</SearchInput>
`;
